### PR TITLE
First push for operator overloading

### PR DESCRIPTION
**Pull request by tavisca-arsalan**
@neilunadkat Please review my code. 
@ytokas Please review my code.
@ytokas Pleaase review the changes pointed out yesterday
@neilunadkat Please review the changes to resource file
@ytokas Please check

### CODE REVIEW

File: 01 OperatorOverloading/OperatorOverloading.Host/HostProgram.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    class HostProgram
+    {
+        static void Main(string[] args)
+        {
+            double temporaryAmount;
+            string temporaryCurrency;
+
+            //Take the input from user for first money object
+            Console.WriteLine("Enter amount for first money object:");
+            temporaryAmount = Convert.ToDouble(Console.ReadLine());
```
Comment (tavisca-arsalan) [inline 01 OperatorOverloading/OperatorOverloading.Host/HostProgram.cs]: use double.tryParse

File: 01 OperatorOverloading/OperatorOverloading.Host/HostProgram.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            Console.WriteLine("Enter currency(in upper case) for first money object:");
+            temporaryCurrency = Convert.ToString(Console.ReadLine());
```
Comment (tavisca-arsalan) [inline 01 OperatorOverloading/OperatorOverloading.Host/HostProgram.cs]: no need to convert string

File: 01 OperatorOverloading/OperatorOverloading.Model/AmountException.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OperatorOverloading.Model
+{
+    public class AmountException:Exception
```
Comment (tavisca-arsalan) [inline 01 OperatorOverloading/OperatorOverloading.Model/AmountException.cs]: make it serializable

File: 01 OperatorOverloading/OperatorOverloading.Model/AmountException.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        public string EMessage { get; set; }
```
Comment (tavisca-arsalan) [inline 01 OperatorOverloading/OperatorOverloading.Model/AmountException.cs]: no need for Emessage, Exception has a Message property

File: 01 OperatorOverloading/OperatorOverloading.Model/Money.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        public Money(double amount, string currency)
+        {
+            Amount = amount;
+            Currency = currency;
+        }
+        public static Money operator +(Money m1, Money m2)
+        {
+            double amount;
+            string currency;
+            if (m1.Currency.Equals(m2.Currency))
```
Comment (tavisca-arsalan) [inline 01 OperatorOverloading/OperatorOverloading.Model/Money.cs]: check for case sensitivity

File: 01 OperatorOverloading/OperatorOverloading.Model/Money.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            {
+                if (m1.Amount==double.MaxValue || m1.Amount < 0 || m2.Amount==double.MaxValue || m2.Amount < 0)
```
Comment (tavisca-arsalan) [inline 01 OperatorOverloading/OperatorOverloading.Model/Money.cs]: not required

File: 01 Operator_Overloading/01 OperatorOverloading/OperatorOverloading.Host/Program.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            try
+            {
+
+                Console.WriteLine("Enter amount for first money object:");
+                while (double.TryParse(Console.ReadLine(), out  temporaryAmount) == false)
+                {
+                    Console.WriteLine("Please enter proper amount for first money object:");
+                }
+                Console.WriteLine("Enter currency for first money object:");
+                temporaryCurrency = Console.ReadLine();
```
Comment (ytokas) [inline 01 Operator_Overloading/01 OperatorOverloading/OperatorOverloading.Host/Program.cs]: What if currency is incorrect? Should ask for it again?

File: 01 Operator_Overloading/01 OperatorOverloading/OperatorOverloading.Model/Money.cs [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+               }
+        }
+
+       
+        public Money(double amount, string currency)
+        {
+            Amount = amount;
+            Currency = currency;
+        }
+        public static Money operator +(Money m1, Money m2)
```
Comment (ytokas) [inline 01 Operator_Overloading/01 OperatorOverloading/OperatorOverloading.Model/Money.cs]: Variable names should be meaningful. Their name should tell what they are.

**Pull request merged**